lighthouse-core/lib/network-recorder.js

```diff
@@ -341,10 +341,16 @@ class NetworkRecorder extends EventEmitter {
     const records = networkRecorder.getRecords().filter(record => record.isValid);
 
     // create a map of all the records by URL to link up initiator
-    const recordsByURL = new Map();
+    const possiblePrefetchRecords = new Map();
+    const typedRecords = new Map();
     for (const record of records) {
-      if (recordsByURL.has(record.url)) continue;
-      recordsByURL.set(record.url, record);
+      if (record.type === NetworkRequest.TYPES.Other) {
+        if (!possiblePrefetchRecords.has(record.url)) {
+          possiblePrefetchRecords.set(record.url, record);
+        }
+      } else if (!typedRecords.has(record.url)) {
```

Comment:
I wonder if we should start taking the same approach here as in page-dependency-graph and not set the initiator if it was ambiguous on the `typedRecords`

Reply:
Possibly. I can take a look in a follow-up PR.

I did look a bit into resolving ambiguities in page-dependency-graph, and they all seemed to increase the errors in simulation. Even when I did:

```
const parentCandidates = (networkNodeOutput.urlToNodeMap.get(initiator) || [])
    .filter(candidate => candidate.startTime < node.startTime);
```

Though there are cases that I would expect to be resolvable. For example, sometimes gpt.js is loaded inside of the top window and inside of iframes. But, again, some of the naive filters I wrote increased the amount of error in simulation so more work would be needed to investigate. (Which is out of scope of this PR but I could look into at some point)